fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/rdf/JcrRdfTools.java

```diff
@@ -405,8 +407,19 @@ private Resource getSkolemizedResource(final IdentifierConverter<Resource, Fedor
         final AnonId id = resource.asResource().getId();
 
         if (!skolemizedBnodeMap.containsKey(id)) {
-            final Node orCreateNode = jcrTools.findOrCreateNode(session, skolemizedId());
+            final String pid = pidMinter.mintPid();
```

Comment:
Because that would tag /wellknown/genid as well, and then all the blank nodes would appear on the main page (instead of under /wellknown/genid). I too wanted jcrTools.findOrCreate() for parent path /wellknown/genid to be called once, but can't think of a way to implement this alternatively in a clean fashion (this method does not explicitly get called in a loop). One option I did originally have in mind was to not use the existing method in AbstractService#tagHierarchyWithPairtreeMixin(), and string tokenize the path, but this approach doesn't seem good to me (split/join on the way in and then splitting again; not using the existing functionality, etc).